candid_rust/tests/assets/examples/linkedup.rs

```diff
@@ -0,0 +1,34 @@
+type UserId_2 = UserId;
+type UserId = principal;
```

Comment:
Do we type check the generated code? We don't support principal yet. The parser is likely to treat it as a variable. If we type check the code, it will probably fail.

Reply:
We don't _type check_ the generated code (as it can be part of another module, but we do syntax checking in tests to make sure it's proper rust.